vtex/node-vtex-api
# Feature/return 429 for specific error type

#### What is the purpose of this pull request?

Create an error type for events that returns status code 429.

#### What problem is this solving?

Avoiding circuit break on retry because all error types return status code 500.

#### How should this be manually tested?

Linking an app that listen to events, linking this version of node-vtex-api in it, importing `TooManyRequestsError` from `@vtex/api` and throwing this error type in one of the event middlewares. Then see if the status code returned is 429.

#### Types of changes
